velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -381,7 +413,27 @@ class DecimalAverageAggregate : public DecimalAggregate<TUnscaledType> {
   }
 };
 
-exec::AggregateRegistrationResult registerAverage(const std::string& name) {
+template <typename TInput, typename TAccumulator, typename TResult>
+static std::unique_ptr<exec::Aggregate> getAggregateFunction(
+    AggregateType aggregateType,
+    const TypePtr& resultType) {
+  if (aggregateType == AggregateType::kAvg) {
+    return std::make_unique<
+        AverageAggregate<TInput, TAccumulator, TResult, AggregateType::kAvg>>(
+        resultType);
+  } else if (aggregateType == AggregateType::kGeoMean) {
+    return std::make_unique<AverageAggregate<
+        TInput,
+        TAccumulator,
+        TResult,
+        AggregateType::kGeoMean>>(resultType);
+  }
+  return nullptr;
+}
+
+exec::AggregateRegistrationResult registerAverage(
```

Comment:
You could also make registerAverage templatized on AggregateType to avoid writing getAggregateFunction. The template variable can be directly used in the make_unique<> statement.